### ISSUE TITLE
Doesn't load. Just crashes.

### ISSUE THREAD

**Issue by FailSafeNow** [apmastering/AP-Mastering-Compressor#1]
Unhandled exception 0xC0000005 at 00007FFF626B3020 reading from 0000000000000000

CS:RIP=33:00007FFF626B3020 SS:RSP=2B:000000001ABDFDD0 RBP=0000000000000000
Flags=0000000000010282 MXCSR=0000000000001F80
DS =000000000000002B ES =000000000000002B FS =0000000000000053 GS =000000000000002B
RAX=0000000000000000 RBX=000000000922C4A0 RCX=000000000922C4B0 RDX=0000000000000000
RSI=000000000922C4A8 RDI=000000000922C4A8
R8 =00000000000002E1 R9 =0000000000000040 R10=00000000000002C0 R11=00000000666314FC
R12=0000000000000000 R13=0000000000000000 R14=0000000000000000 R15=0000000000000000
Stack Trace:
00007FFF626B3020 (MSVCP140): Thrd_yield + 0x110
00007FFEBD3AC6B7 (APComp): InitDll + 0x14337
00007FFEBD454027 (APComp): InitDll + 0xbbca7
00007FFEBD3AD58C (APComp): InitDll + 0x1520c
00007FFF67AA9333 (ucrtbase): recalloc + 0xa3
00007FFF6969257D (KERNEL32): BaseThreadInitThunk + 0x1d
00007FFF6A4EAF28 (ntdll): RtlUserThreadStart + 0x28

**Comment by apmastering** [comment]
Do you have the current version of the Microsoft c++ runtime installed?

**Comment by FailSafeNow** [comment]
Yup. Microsoft Visual C++ 2015-2022 Redistributable (x86) and (x64) 14.40.33810.

**Comment by apmastering** [comment]
part of the reason why i made it open source is so others could look at the code and point out why it's getting weird crashes on windows. i don't use windows and don't know what the problem is. on mac it works just fine.

**Comment by FailSafeNow** [comment]
I have some news, and it surprises me. I have no experience whatsoever building VST files or using JUCE... but with the help of ChatGPT, I _actually_ got it to work.

It was missing some references to image files, and some of the code in APCommon.cpp and APCommon.h wasn't quite right. But to my amazement, with some modifications, I actually compiled the plugin, and it's running. I don't know how to use all of the parameters — I'd love a detailed guide on them. But it works.

In Projucer, the list of files needed to include bgflat.png and knobflat.png. And with the modifications to APCommon.cpp and APCommon.h which I am uploading, the plugin was able to build, and it works.

There were a bunch of warnings about "possible loss of data" during the build, regarding conversions of doubles to other variable types. But it runs.

[Source.zip](https://github.com/user-attachments/files/16264747/Source.zip)

**Comment by apmastering** [comment]
good work! it sounds like you were compiling the new skin code with the old juicer file or something because i recently uploaded a new flat skin that i introduce in todays video. In terms of AP common, that's my db to decimal and back functions.... feel free to change what you want but if you changed the conversion there it might lead to unexpected results with the audio.... if thats the make or break part then if might have to do with how infinity vales are handled on windows... thanks for effort!

**Comment by FailSafeNow** [comment]
Will you be exporting a new version of the plugin with the fixes?

**Comment by apmastering** [comment]
i'm on the road right now so don't have my computer, just phone. i'll look when i get back! :-)

**Comment by apmastering** [comment]
btw i can't see any pull requests from you

**Comment by FailSafeNow** [comment]
Didn't even know what one was until now.

**Comment by apmastering** [comment]
feel free to send me the code via email to compressor@apmastering.com if you'd can't work out how to do a pull request

**Comment by apmastering** [comment]
oh just saw you sent a zip in your previous message. will look now

**Comment by apmastering** [comment]
OK so I guess the main issue you found was that on windows I need to include the "limits" library... not sure why and I can't test to see if this solved the issue for your type of windows, but I've included that anyway, and let's see if it works!

ive just uploaded a new windows binary to the google drive folder, if you are up for grabbing it and giving it a go, that would be fantastic!

**Comment by FailSafeNow** [comment]
I think you put more in the zip than you meant to.

Anyhow, the debug version of the plugin loads, but the regular one doesn't. I'll have to check on that.

**Comment by FailSafeNow** [comment]
Ok, so the release build doesn't work. I tried building it myself from the latest source, and it didn't work for me either. The debug version of the plugin, which comes in at about 21 or 22 MB, can run. Maybe it makes sense to just release the debug version of the plugin as a stopgap, until whatever other issues can be resolved.

**Comment by apmastering** [comment]
If the debug version that you compile yourself works for you but not the binary I uploaded, this sounds like it could be windows SmartScreen block or an overly enthusiastic antivirus software or something like this interfering. Yesterday I did a big code overhaul and this latest version might totally fix the problem you were having but otherwise I haven't been able to reproduce this error on windows. On all three platforms, Windows, Mac and linux, everything is running rock solid now with the VST3 and the AU passes validation and seems totally solid too. So for now I'm going to close this particular issue because I am unable to reproduce and the update may have fixed it anyway.

But if you have made sure its not an antivirus or windows blocking the VST, and you still get crashes with the new windows binary, then please do submit a new crash report and I'll try again to reproduce it!

**Comment by apmastering** [comment]
I think this issue was caused by the Microsoft c++ runtime all along actually.

Another user had the same error so I compiled a new version statically linking the c++ runtime and it works fine for him now.

It seems, despite the fact that microsoft includes this runtime as part of all windows systems as of version 10, it is sometimes a weird version etc, so statically linking it solved the problem.... unless your specific report was because of something else. By anyone, if you haven't got totally bored with the plugin by now, the static version in the google drive SHOULD work ha ha
